File: app/src/main/java/com/derlados/computerconf/Constants/CompatParam.java
package com.derlados.computerconf.Constants;

public class CompatParam {

    public static class Cpu {
        public final static String SOCKET = "Тип сокета";
    }

    public static class Ram {
        public final static String TYPE_MEMORY = "Тип памяти";
        public final static String SIZE_MEMORY = "Объем памяти";
    }

    public static class PowerSupply {
        public final static String POWER = "Мощность";
    }

    public static class Case {
        public final static String FROM_FACTOR = "Поддерживаемые материнские платы";
    }

    public static class Motherboard {
        public final static String SOCKET = "Сокет";
        public final static String TYPE_MEMORY = "Тип оперативной памяти";
        public final static String MAX_SIZE_MEMORY = "Максимальний объем памяти";
        public final static String FROM_FACTOR = "Форм-фактор";
        public final static String COUNT_RAM = "Количество разъемов, шт.";
    }

    public static class Gpu {
        public final static String POWER = "Минимальная рекомендованная мощность блока питания";
    }

}

File: app/src/main/java/com/derlados/computer_conf/constants/CompatParam.kt
package com.derlados.computer_conf.Constants;

class CompatParam {

    class Cpu {
        companion object {
            const val SOCKET: String = "Тип сокета";
        }
    }

    class Ram {
        companion object {
            const val TYPE_MEMORY = "Тип памяти";
            const val SIZE_MEMORY = "Объем памяти";
        }
    }

    class PowerSupply {
        companion object {
            const val POWER = "Мощность";
        }
    }

    class Case {
        companion object {
            const val FROM_FACTOR = "Поддерживаемые материнские платы";
        }
    }

    class Motherboard {
        companion object {
            const val SOCKET = "Сокет"
            const val TYPE_MEMORY = "Тип оперативной памяти"
            const val MAX_SIZE_MEMORY = "Максимальний объем памяти"
            const val FORM_FACTOR = "Форм-фактор"
            const val COUNT_RAM = "Количество разъемов, шт."
            const val PORTS = "Внутренние разъемы и порты"
        }
    }

    class Gpu {
        companion object {
            const val POWER = "Минимальная рекомендованная мощность блока питания"
        }
    }
}
